src/cpuinfo_x86.c

```diff
@@ -689,10 +689,25 @@ X86Microarchitecture GetX86Microarchitecture(const X86Info* info) {
       case CPUID(0x06, 0x5E):
         // https://en.wikipedia.org/wiki/Skylake_(microarchitecture)
         return INTEL_SKL;
+      case CPUID(0x06, 0x66):
+        // https://en.wikipedia.org/wiki/Cannon_Lake_(microarchitecture)
+        return INTEL_CNL;
+      case CPUID(0x06, 0x7E):
+        // https://en.wikipedia.org/wiki/Ice_Lake_(microprocessor)
+        return INTEL_ICL;
       case CPUID(0x06, 0x8E):
+        if (info->stepping > 10) {
+          // https://en.wikipedia.org/wiki/Whiskey_Lake_(microarchitecture)
+          return INTEL_WHL;
+        }
```

Comment:
Please add to 0x06, 0x8E
```cpp
if (9 == info->stepping) {
           // https://en.wikipedia.org/wiki/Kaby_Lake
           return INTEL_KBL;
} else if (10 == info->stepping) {
           // https://en.wikipedia.org/wiki/Coffee_Lake
           return INTEL_CFL;
} else if (11 == info->stepping) {
          // https://en.wikipedia.org/wiki/Whiskey_Lake_(microarchitecture)
          return INTEL_WHL;  
} else {
  return X86_UNKNOWN;
}
```

ref:
* https://en.wikichip.org/wiki/intel/microarchitectures/kaby_lake#CPUID
Family 6 Model 142 Stepping 9

* https://en.wikichip.org/wiki/intel/microarchitectures/coffee_lake#CPUID
Family 6 Model 142 Stepping 10

* https://en.wikichip.org/wiki/intel/microarchitectures/whiskey_lake#CPUID
Family 6 Model 142 Stepping 11

Reply:
@Mizux Is this ok?